### ISSUE TITLE
Starting May 5th, you must let us know why your app requires broad storage access

### ISSUE THREAD

**Issue by diegolmello** [ivpusic/react-native-image-crop-picker#1571]
### Version
Tell us which versions you are using: 

- react-native-image-crop-picker v0.31.1
- react-native v0.63.4

### Platform
Tell us to which platform this issue is related

- Android

From Google Play inbox:
> We've detected that your app contains the requestLegacyExternalStorage flag in the manifest file of 1 or more of your app bundles or APKs.

> Developers with apps on devices running Android 11+ must use Scoped Storage to give users better access control over their device storage. To release your app on Android 11 or newer after May 5th, you must either:

> Update your app to use more privacy friendly best practices, such as the Storage Access Framework or Media Store API
Update your app to declare the All files access (MANAGE_EXTERNAL_STORAGE) permission in the manifest file, and complete the All files access permission declaration in Play Console from May 5th
Remove the All files access permission from your app entirely
For apps targeting Android 11, the requestLegacyExternalStorage flag will be ignored. You must use the All files access permission to retain broad access.

> Apps requesting access to the All files access permission without a permitted use will be removed from Google Play, and you won't be able to publish updates.

Do we have a workaround for that?

Thanks!

Love react-native-image-crop-picker? Please consider supporting our collective:
👉  https://opencollective.com/react-native-image-crop-picker/donate

**Comment by phatlaunchdeck** [comment]
I got this message too and would like to know if anyone can come up with a solution

**Comment by aj3012** [comment]
+1

**Comment by jacquesdev** [comment]
I suspect that some people get that warning after using the suggested fix here https://github.com/ivpusic/react-native-image-crop-picker/issues/1449#issuecomment-724638956, not sure if it affects everyone though?

**Comment by dolphinflow86** [comment]
+1

**Comment by andresribeiro** [comment]
+1

**Comment by Steffi3rd** [comment]
> I suspect that some people get that warning after using the suggested fix here [#1449 (comment)](https://github.com/ivpusic/react-native-image-crop-picker/issues/1449#issuecomment-724638956), not sure if it affects everyone though?

Yes, indeed. (for me)

**Comment by kursonix** [comment]
I do research about it and the waring is applicable to case when you are targeting level 30 Android 11 I think. If you target level 29 it should be ok right now, bu in November 2021 it is required to use Api level 30 for existing app if you want to place them in google play store. Google play policy timeline https://developer.android.com/distribute/play-policies

**Comment by saulowilson** [comment]
Is there some deadline when the lib will be updated?

**Comment by JDGarner** [comment]
In case it helps anyone else, I switched to using https://github.com/rnmods/react-native-document-picker for now, it can be used as an Image/Video picker by passing in the correct types, e.g:
```
DocumentPicker.pickMultiple({
      type: [DocumentPicker.types.images, DocumentPicker.types.video],
    });
```
And doesn't require the use of `requestLegacyExternalStorage` to pick images from the gallery/other apps

**Comment by thuynt99** [comment]
Have any plan add scope storage for android 11?

**Comment by ghasemikasra39** [comment]
Any solution?

**Comment by AathikaMSB-Rently** [comment]
Any updates on this issue ?

**Comment by mkamals** [comment]
Waiting for solution to this issue.

**Comment by thuynt99** [comment]
Waiting for solution to this issue.

**Comment by KochMario** [comment]
Any updates? 🤔
@ivpusic I see that you started working on this on July 19th. Did you make any progress so far?

**Comment by ivpusic** [comment]
would be cool if somebody from the community has some time to finish support for android 11+

**Comment by KochMario** [comment]
Since November is only a couple of days ahead I decided it's too risky to keep hoping for an update before then and therefore I've combined `react-native-image-picker` and `react-native-image-crop-picker` and implemented some sort of patchy solution with both of them.

Since `react-native-image-picker` is compatible with targetSdkVersion >= 30 (and therefore Android 11) I use that to access storage and capture images/videos with the camera. For additional cropping of the selected or captured photos I'm then additionally using `react-native-image-crop-picker` (since `react-native-image-picker` does not provide that functionality).

In my opinion, this works pretty decently (for the time being), since I've managed to cover the entire workflow I had before. The production release with that update was just accepted and published to Google Play Store.

If anyone is interested in the corresponding code, let me know and I'm happy to share it here.

**Comment by kristijantomic** [comment]
@KochMario i would like to see the code. Thx!

**Comment by SudoPlz** [comment]
@PaperMonster does that mean the issue is fixed on 0.36.4 ? Can the maintainer of this library shed some light?
Did someone work on fixing this?
Thanks 🙏

**Comment by SudoPlz** [comment]
Ok to answer my question, this library is still using the deprecated  `WRITE_EXTERNAL_STORAGE` permission and thus won't work on any app that uses Android API 30 and above (which since we're on November, is required to upload a binary to the store).

https://github.com/ivpusic/react-native-image-crop-picker/blob/master/android/src/main/java/com/reactnative/ivpusic/imagepicker/PickerModule.java#L200

@ivpusic can you please confirm?

Thanks

**Comment by Hesowcharov** [comment]
@SudoPlz Hi! I can't absolutely agree with you because later piece of code https://github.com/ivpusic/react-native-image-crop-picker/blob/2b95b88f0b45025cccea0d39be582ff4cf311d1e/android/src/main/java/com/reactnative/ivpusic/imagepicker/PickerModule.java#L225-L233
conditionally remove `WRITE_EXTERNAL_STORAGE` for android 11 🤔 . Do you think it is still an issue if that code just refers to `WRITE_EXTERNAL_STORAGE` permission?

**Comment by PaperMonster** [comment]
@SudoPlz I released an app with version 0.36.4 + target API level 30 + `WRITE_EXTERNAL_STORAGE` in AndroidManifest last week on Google Play. It's been 4 days and there's no issue reported yet. At least I can confirm image picking and cropping works as expected on Samsung Galaxy Note 20 + Android 11.

**Comment by SudoPlz** [comment]
I can confirm that conditionally removing the `WRITE_EXTERNAL_STORAGE` permission stops our app from crashing too.
Looks like that was all it took.

In our app things were a bit more complicated though, because an external dependency (instabug) specified a `maxSdkVersion` to the `WRITE_EXTERNAL_STORAGE` without us knowing about it (which resulted in weird behavior).

All good after all.

**Comment by joy-betterhalf** [comment]
@PaperMonster @SudoPlz 

Can you guys please confirm if the play store has accepted your app after reviewing it? Because they have been rejecting our app and the reason for it is Scope storage access.

Even though our app doesn't ask for `MANAGE_EXTERNAL_STORAGE` permission they keep on rejecting our app saying that we are requesting `MANAGE_EXTERNAL_STORAGE` permission or not following best practicing. 

Please check the below screenshot where clearly I can't see `MANAGE_EXTERNAL_STORAGE` permission as part of Manifest but `WRITE_EXTERNAL_STORAGE` is there which is conditionally handled by **react-native-image-crop-picker** and permission will not ask for Android 11. So, I don't understand the issue here and I don't understand why they are rejecting the app. 

<img width="1595" alt="Screenshot 2021-12-23 at 1 19 20 AM" src="https://user-images.githubusercontent.com/82091935/147147508-7a6d0377-0332-4512-95c2-f101b7e54d8e.png">


@ivpusic Please help

Thank you.

**Comment by PaperMonster** [comment]
@joy-betterhalf Yes, I can confirm mine. My app has `WRITE_EXTERNAL_STORAGE` but doesn't have `MANAGE_EXTERNAL_STORAGE`. In my case, `WRITE_EXTERNAL_STORAGE` is also needed for `@react-native-community/cameraroll` in Android < 10 but I had to write native module for `CameraRoll.save()` for Android >= 10.

Perhaps you need to make sure `requestLegacyExternalStorage=true` is not in AndroidManifest.xml and check other dependencies if any of them uses forbidden API (like what `@react-native-community/cameraroll` is using).

**Comment by SudoPlz** [comment]
Same here, I can confirm mine too, the app has `WRITE_EXTERNAL_STORAGE` but doesn't have `MANAGE_EXTERNAL_STORAGE`.

We don't use `requestLegacyExternalStorage` at all now, but I bet it wouldn't be a problem if we did.
As @PaperMonster said make sure you check your merged manifest, and see what your other dependencies add to the final manifest. (We actually did have 1 dependency that messed with the `WRITE_EXTERNAL_STORAGE` dependency without our knowledge (instabug).

**Comment by joy-betterhalf** [comment]
@PaperMonster @SudoPlz 

Thank you guys for your quick response. I really appreciate your help. This information will help me solve my issue and hopefully Play store will accept our app. 🤞

**Comment by retyui** [comment]
> I have been able to use the package as well without any issues. I am patching it though to update a number of the dependency versions and to resolve errors/warnings provided by Android Studio. Patch code in txt file below..
> 
> [react-native-image-crop-picker+0.36.4.patch.txt](https://github.com/ivpusic/react-native-image-crop-picker/files/7331459/react-native-image-crop-picker%2B0.36.4.patch.txt)

Thanks your fix is working for me!

**Comment by RodolfoGS** [comment]
Any update about this, it's still ocurring with **0.40.0** + **react-native 0.72.4**
Error code: `E_NO_LIBRARY_PERMISSION`

**Comment by RodolfoGS** [comment]
I created a PR that fix it https://github.com/ivpusic/react-native-image-crop-picker/pull/1973
